third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -44,6 +44,50 @@ int getWmmaVersion(StringRef archGen) {
   return 0;
 }
 
+// Check if the result of current tl.dot is used as the operand(0)
+// of another tl.dot
+bool isChainDotHead(tt::DotOp &dotOp) {
```

Comment:
In these two functions, use `DotOpInterface` instead of hardcoded `tt::DotOp` so it works for `DotScaledOp` too. Also typically for friendly named `Op`s (that is, `tt::DotOp`, not `Operation *`), we don't pass as a reference; we directly pass as a value because `Op`s are just a wrapper of `Operation *`.